src/plugins/discover/public/application/angular/discover.js

```diff
@@ -700,7 +700,11 @@ function discoverController($route, $scope, Promise) {
 
   async function setupVisualization() {
     // If no timefield has been specified we don't create a histogram of messages
-    if (!getTimeField() || $scope.state.hideChart) return;
+    if (!getTimeField() || $scope.state.hideChart) {
+      if ($scope.volatileSearchSource.getField('aggs'));
+      $scope.volatileSearchSource.removeField('aggs');
+      return;
+    }
```

Comment:
Removing `aggs` field for the case the user is hiding the chart, and a request with `aggs` was executed before